core/utils.h

```diff
@@ -1946,12 +1946,12 @@ notify(syslog_event_type_t priority, bool internal, bool synch,
 #define SYSLOG_INTERNAL_CRITICAL_ONCE(...) \
     DODEBUG_ONCE(SYSLOG_INTERNAL_CRITICAL(__VA_ARGS__))
 
-#define MSG_COULDNT_SYNCH_WITH_ALL_THREADS_CODE 42
-
-#define fatal_error(id, sub, ...) \
+#define REPORT_FATAL_ERROR_AND_EXIT(dcontext, msg_id, arg_count, ...) \
     do { \
-      SYSLOG_COMMON(false, SYSLOG_CRITICAL, id, sub, __VA_ARGS__); \
-      exit_process_syscall(MSG_##id##_CODE); \
+        SYSLOG_COMMON(false, SYSLOG_CRITICAL, msg_id, arg_count, \
+                      __VA_ARGS__); \
```

Comment:
Potentially missing features from only calling notify() on a fatal error are a core dump (ldmp for -dumpcore_mask) and for static libs invoking the app's crash handler for more diagnostics -- seems fine to not have for now, just sthg to keep in mind, maybe a comment

Reply:
I've added a comment about potential future additions.